### PR TITLE
Add dotnet-format tool

### PR DESCRIPTION
**Pull request by sungam3r**
Inspired by https://github.com/EasyNetQ/EasyNetQ/pull/1186

### CODE REVIEW

**Comment by sungam3r** [comment]
@Shane32 I am sure this will remove a lot of the problem of unnecessary file changes.

File: src/Directory.Build.props [hunk 1]
```diff
@@ -18,7 +18,7 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0" PrivateAssets="All" Condition="'$(IsPackable)' == 'true'"/>
```
Comment (sungam3r) [inline src/Directory.Build.props:21]: https://github.com/dotnet/sourcelink/issues/91#issuecomment-400787926

Comment (sungam3r) [reply src/Directory.Build.props]: It turns out that this was a warning not affecting the correct formatting. 🤦

**Comment by sungam3r** [comment]
Finally done!

**Review by Shane32 — APPROVED** [review]

**Pull request merged**